mlflow/sklearn/utils.py

```diff
@@ -615,9 +624,30 @@ def _create_child_runs_for_parameter_search(cv_estimator, parent_run, child_tags
     # the seed estimator and update them with parameter subset specified
     # in the result row
     base_params = seed_estimator.get_params(deep=should_log_params_deeply)
-
     cv_results_df = pd.DataFrame.from_dict(cv_estimator.cv_results_)
-    for _, result_row in cv_results_df.iterrows():
+
+    if max_tuning_runs is None:
+        cv_results_best_n_df = cv_results_df
+    else:
+        rank_column_name = "rank_test_score"
+        if rank_column_name not in cv_results_df.columns.values:
+            rank_column_name = first_custom_rank_column(cv_results_df)
+            warnings.warn(
+                "Top {} child runs will be created based on ordering in {} column.".format(
+                    max_tuning_runs, rank_column_name,
+                )
+                + " You can choose not to limit the number of child runs created by"
+                + " setting `max_tuning_runs=None`."
+            )
+        cv_results_best_n_df = cv_results_df.nsmallest(max_tuning_runs, rank_column_name)
+        # Log how many child runs will be created vs omitted.
+        rest = len(cv_results_df) - max_tuning_runs
+        if rest > 0:
+            _logger.info(
+                "Logging the %s best runs, %s others will be omitted.", max_tuning_runs, rest,
+            )
```

Comment:
Nit: This message becomes a bit odd when `max_tuning_runs` is zero or one.

"Logging the best 0 runs, 15 others will be omitted."

"Logging the best 1 runs, 14 others will be omitted."

Can we handle these cases?